crates/extensions/tedge_http_ext/src/lib.rs

```diff
@@ -13,30 +13,41 @@ use actor::*;
 use tedge_actors::Concurrent;
 use tedge_actors::ServerActorBuilder;
 use tedge_actors::ServerConfig;
+use tedge_config::TEdgeConfig;
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct HttpActor {
     config: ServerConfig,
+    tls_client_config: rustls::ClientConfig,
 }
 
 impl HttpActor {
-    pub fn new() -> Self {
-        HttpActor::default()
+    pub fn new(tedge_config: &TEdgeConfig) -> Self {
+        Self {
+            config: <_>::default(),
+            tls_client_config: tedge_config.cloud_client_tls_config(),
+        }
     }
 
     pub fn builder(&self) -> ServerActorBuilder<HttpService, Concurrent> {
-        ServerActorBuilder::new(HttpService::new(), &self.config, Concurrent)
+        ServerActorBuilder::new(
+            HttpService::new(self.tls_client_config.clone()),
+            &self.config,
+            Concurrent,
+        )
     }
 
-    pub fn with_capacity(self, capacity: usize) -> Self {
+    pub fn with_capacity(self, capacity: usize, tedge_config: &TEdgeConfig) -> Self {
```

Comment:
That's another reason for not taking the `&TEdgeConfig` in the constructor.

Reply:
Ah, that's a bug, we don't need to recreate `self` here, I'll fix it and then we'll only have the config in the one place.